File: 29.cpp
/*思路：先排序，后取前n个数字*/

vector<int> GetLeastNumbers_Solution(vector<int> input, int k) 
{
    vector<int> result;  
    int len = input.size();  
    if(len !=0&&!(len<k)) 
    {  
        sort(input.begin(),input.end());  
        for(int i = 0;i<k;++i)
        {     
            result.push_back(input[i]);   
        }  
        return result;
    }
    else 
        return result;
}

File: 29.py
# -*- coding:utf-8 -*-
class Solution:
    def GetLeastNumbers_Solution(self, tinput, k):
        # write code here
        if k > len(tinput) or k == 0:
            return []
        newinput = sorted(tinput)
        result = []
        for i in range(len(newinput)):
            if i != k-1:
                result.append(newinput[i])
            else:
                result.append(newinput[i])
                return result